tests/core/test_macros.py

```diff
@@ -663,3 +663,169 @@ def test_macro_first_value_ignore_respect_nulls(assert_exp_eq) -> None:
         "SELECT FIRST_VALUE(@test(x) RESPECT NULLS) OVER (ORDER BY y) AS column_test"
     )
     assert_exp_eq(evaluator.transform(actual_expr), expected_sql, dialect="duckdb")
+
+
+@pytest.mark.parametrize(
```

Comment:
There are basically two expected outputs in this test - those that include `QUALIFY` and those that don't. I'd consider refactoring it to compact it a bit because there's a lot of duplication now. We can partition by dialect that supports `QUALIFY` vs not to control the expected output.

Reply:
done

Reply:
redshift is separated out due to its quoting